tests/src/Context/DrupalContext.php

```diff
@@ -27,6 +28,53 @@ public function loggedIn() {
   }
 
   /**
+   * Checks the existence of a content page.
+   *
+   * @param int $title
+   *   The title of the page.
+   *
+   * @throws \Exception
+   *   Thrown when the entity is not found.
+   *
+   * @Then I should have a :type (content )page titled :title
+   */
+  public function assertContentPageByTitle($type, $title) {
+    $type = $this->getBundleFromLabel($type);
+
+    $nodes = \Drupal::entityTypeManager()->getStorage('node')
+      ->loadByProperties(
+        ['title' => $title, 'type' => $type]
+      );
+    $actual = reset($nodes);
+    if (empty($actual)) {
+      throw new \Exception("Entity titled '$title' was not found.");
+    }
+  }
+
+  /**
+   * Returns the machine name of a node bundle given their Bundle Label.
+   *
+   * @param string $bundle_label
+   *   The label of the bundle.
+   *
+   * @return string
+   *   The machine name of the bundle.
+   *
+   * @throws \Exception
+   *   Thrown when the bundle type is not found.
+   */
+  public function getBundleFromLabel($bundle_label) {
+    foreach (NodeType::loadMultiple() as $type) {
+      if ($bundle_label == $type->label()) {
+        return $type->id();
+      }
+    }
```

Comment:
Retrieving all node types and looping over them to find the right one is not very efficient. I understand the philosophy though, we will only have very few node types in this project so this won't take much time at all. Still it would be better to simply to a query that retrieves the node type by label.

A second point is that this is hard coded on node types. We will need to do the same for RDF types, and in the future even more types. If we would make this generic now, then we don't need to duplicate code and we can save time in the future.

We can even take it further. Since an entity type in itself is _also an entity_, we can make this more generic: let it match on _any entity_ that matches a certain label and entity type. Since the node title is also a label this code can then also be used to look up a node with a certain title. We can hit three birds with one stone!

That would make the code that does this in `assertContentPageByTitle()` above redundant, after all it first retrieves the node type by looking up the entity that matches the "type" label, and then the node by looking up the entity that matches the "title" label. That would reduce the code above to the following:

```
$bundle = $this->getEntityFromLabel('node_type', $type);
$node = $this->getEntityFromLabel('node', $label, $bundle);
```